src/tir/schedule/primitive/compute_at.cc

```diff
@@ -422,19 +422,27 @@ std::pair<Var, BlockVarDomainInfo> SolveBlockVarDomain(const arith::IntSet& prov
       var_dom = arith::IntSet::Interval(required_min, required_max);
       var_bound = arith::IntSet::Interval(0, dim_max);
     } else {
-      arith::PVar<PrimExpr> p_f;
-      if ((floordiv(p_v, p_f)).Match(provided_min)) {
+      arith::PVar<PrimExpr> p_f1, p_f2;
+      if ((floordiv(p_v, p_f1)).Match(provided_min)) {
         // a <= (x // factor) <= b, fac > 0 ==> (a * fac) <= x <= (b * fac + fac - 1)
-        PrimExpr fac = p_f.Eval();
+        PrimExpr fac = p_f1.Eval();
         if (analyzer->CanProveGreaterEqual(fac, 1)) {
           var = p_v.Eval();
           var_dom = arith::IntSet::Interval(required_min * fac,
                                             analyzer->Simplify(required_max * fac + fac - 1));
           var_bound = arith::IntSet::Interval(0, analyzer->Simplify(dim_max * fac + fac - 1));
         }
-      } else if ((floormod(p_v, p_f).Match(provided_min))) {
+      } else if ((floormod(p_v, p_f1).Match(provided_min))) {
         // generally domain of (x % fac) enforce no constraints to domain of x
         return {p_v.Eval(), BlockVarDomainInfo()};
+      } else if ((floordiv(p_f1, p_f2).Match(provided_min))) {
```

Comment:
Yeal, `p_f1.Eval()` is `PrimExpr` but var is `Var`, it could write as `var = Downcast<Var>(p_f1.Eval())`.

Of course, we should first check whether the `p_f1.Eval()` is an instance of VarNode, if it is, we convert it to `Var` safely and end the recursion.